src/System.Runtime.Extensions/tests/System/IO/PathTests.netcoreapp.cs

```diff
@@ -172,5 +172,177 @@ public static void GetFileNameWithSpaces_Unix_Span()
         {
             Assert.Equal("fi  le", new string(Path.GetFileName(Path.Combine("b \r\n ar", "fi  le").AsReadOnlySpan())));
         }
-    }
+
+		[Fact]
+		[PlatformSpecific(TestPlatforms.Windows)]
+		public static void GetFullPath_BasePath_InvalidInput_Windows()
+		{
+			AssertExtensions.Throws<ArgumentNullException>("basePath", () => Path.GetFullPath("", null));
+			AssertExtensions.Throws<ArgumentNullException>("basePath", () => Path.GetFullPath("tmp", null));
+			AssertExtensions.Throws<ArgumentNullException>("basePath", () => Path.GetFullPath(@"c:\git\corefx", null));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath("", @"foo\bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath("tmp", @"foo\bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath(@"C:\git", @"foo\bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath(null, @"foo\bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath(null, @"C:\foo\bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath("C:\\gi\0t", @"c:\foo\bar"));
```

Comment:
This is a question more about the implementation than the tests, but is it correct that these are all throwing the base ArgumentException and that they don't include a parameter name?  Is that something we can improve?

Reply:
Yeah I can add the argument name to the exception being thrown.

Reply:
cc @JeremyKuhne